src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -936,20 +889,14 @@ public void SetCustomProperty(string propertyName, object? propertyValue)
             else
                 activity._id = activity.GenerateHierarchicalId();
 
-            if (links != null)
+            if (links != null && links.Any())
```

Comment:
It'd be better not to use LINQ here.  This could end up allocating an enumerator.  Since we're going to do that any way as part of the LinkedList constructor, it'd be better to just do it once and change the ctor to take an enumerator rather than enumerable, e.g.
```C#
if (links != null)
{
    IEnumerator<Link> e = links.GetEnumerator();
    if (e.MoveNext())
        activity._links = new LinkedList<ActivityLink>(e);
    else
        e.Dispose();
}
```
Then the `LinkedList` ctor would expect the incoming enumerator to already be positioned on the first element, e.g.
```C#
_last =_first = new LinkedListNode<T>(enumerator.Current);
while (enumerator.MoveNext())
{
    _last.Next = new LinkedListNode<T>(enumerator.Current);
    _last = _last.Next;
}
```

Reply:
a good idea will do it.